### ISSUE TITLE
Support amp-carousel to pause after it loops X times

### ISSUE THREAD

**Issue by jasti** [ampproject/amphtml#18911]
The specific FR is for AMPHTML ads where ads policy might mark an ad as spam if it continues to rotate without ever stopping but I can see this extend to use in regular web pages as well. 

Most probable usecase might for the carousel loop just once and then reset to the first image in the carousel. 

CC @aghassemi

**Comment by aghassemi** [comment]
`loop=opt_N` seems like a reasonable thing to add to either existing Carousel or V2

**Comment by nainar** [comment]
There are 2 cases in which we loop:
Case 1:
The developer has specified the `loop` attribute intentionally. In that case this should be easy enough to support. The only constraint is does the validator support having an attribute with & without a value?

Case 2:
This case is more complicated, here if the developer specifies `autoplay` we then force the `loop` property. In that case for ads do we want to `loop` forever or restrict to a few runs. If we restrict to a few runs that would be a breaking change in carousel V1. 

@jasti @aghassemi thoughts?

**Comment by aghassemi** [comment]
Interesting, maybe `autoplay=opt_N_of_loops` makes more sense? After your comment, I don't think restricting manual `loop` to a particular number makes sense.

**Comment by jasti** [comment]
+1 regarding expecting explicit loop configuration in both case 1 and case 2. 
Reg Case 1 que, yes, I think the validator should support both where without a value defaults to current v1 behavior.

**Comment by honeybadgerdontcare** [comment]
@nainar 	The validator can support it having a value and not a value, most likely using `value_regex` if the values are to be limited to some predefined condition and enumerable beyond a small set. Otherwise it could be repeated fields of `value`.

**Comment by nainar** [comment]
I'll draw up a PR with the following plan then (should be doable for V1):
- Support for `autoplay=opt_N_of_loops`
- If autoplay is specified without a number of loops, it will loop infinitely
- If autoplay is specified with a defined number of loops, we will clear autoplay status on the carousel.

Regarding Carousel V2 - I have added this FR into the PRD.

**Comment by nainar** [comment]
@jasti this is FR is in master.

**Comment by jasti** [comment]
Awesome! Would you please close this issue out once the feature is launched and I'll let the issue requestor know?

**Comment by sparhami** [comment]
Closing as this feature is already live.
